src/uct/ib/base/ib_device.c

```diff
@@ -659,21 +659,16 @@ uint8_t uct_ib_to_rnr_fabric_time(double time)
     double avg_ms;
     uint8_t t;
 
-    if (time_ms <= uct_ib_qp_rnr_time_ms[0]) {
-        return 1;
-    }
-
     for (t = 1; t < UCT_IB_FABRIC_TIME_MAX; t++) {
-        if (time_ms <= uct_ib_qp_rnr_time_ms[t]) {
-            avg_ms = (uct_ib_qp_rnr_time_ms[t - 1] +
-                      uct_ib_qp_rnr_time_ms[t]) * 0.5;
+        if (time_ms <= uct_ib_qp_rnr_time_ms[(t + 1) % UCT_IB_FABRIC_TIME_MAX]) {
```

Comment:
maybe use helper var:
rename t -> 'index'
and (t + 1) % UCT_IB_FABRIC_TIME_MAX --> 'next_index'

Reply:
good catch! done